src/store/index.js

```diff
@@ -0,0 +1,129 @@
+    },
+    async transferMoney({ commit }, { name, money }) {
+      const transferInformation = { name, money }
+      await commit('transferMoney', transferInformation)
+    },
+  },
+  mutations: {
+    setUser(state, userInformation) {
+      firebase
+        .auth()
+        .createUserWithEmailAndPassword(
+          userInformation.email,
+          userInformation.password,
+        )
+        .then(() => {
+          state.userStorage = firebase.auth().currentUser
+          state.userStorage.updateProfile({
+            displayName: userInformation.name,
+          })
+          db.collection('users').add({
+            Name: userInformation.name,
+            Email: userInformation.email,
+            Password: userInformation.password,
+            Money: state.firstMoney,
+          })
+        })
+        .catch((error) => {
+          alert(error)
+        })
+    },
+    login(state, loginInformation) {
+      firebase
+        .auth()
+        .signInWithEmailAndPassword(
+          loginInformation.email,
+          loginInformation.password,
+        )
+        .then(() => {
+          state.loginUserName = firebase.auth().currentUser.displayName
+          db.collection('users')
+            .get()
+            .then((querySnapshot) => {
+              querySnapshot.docs.forEach((doc, index) => {
+                if (
+                  querySnapshot.docs[index].data().Name === state.loginUserName
+                ) {
+                  state.loginUserMoney = querySnapshot.docs[index].data().Money
+                }
+              })
+            })
+        })
+        .catch((error) => {
+          alert(error)
+        })
+    },
+    transferMoney(state, transferInformation) {
+      // 送金する相手のデータを取得
+      db.collection('users')
+        .get()
+        .then((querySnapshot) => {
+          querySnapshot.docs.forEach((doc, index) => {
+            if (
+              querySnapshot.docs[index].data().Name === transferInformation.name
+            ) {
+              state.targetUserID = querySnapshot.docs[index].id
+              state.targetUserMoney = querySnapshot.docs[index].data().Money
+            }
+            if (querySnapshot.docs[index].data().Name === state.loginUserName) {
+              state.loginUserID = querySnapshot.docs[index].id
+            }
+          })
+          state.loginUserMoney -= transferInformation.money
+          state.targetUserMoney += transferInformation.money
+          const loginUserRef = db.collection('users').doc(state.loginUserID)
+          const targetUserRef = db.collection('users').doc(state.targetUserID)
+          //送金処理
+          db.runTransaction(async function (tr) {
+            await tr.get(loginUserRef)
+            await tr
+              .get(targetUserRef)
+              .then(() => {
+                tr.update(loginUserRef, { Money: state.loginUserMoney })
+              })
+              .then(() => {
+                tr.update(targetUserRef, {
```

Comment:
awaitを使用すると、処理をthenの外に出せネストを防げます。

```suggestion
            await tr.get(loginUserRef)
            await tr.get(targetUserRef)
            await tr.update(loginUserRef, { Money: state.loginUserMoney })
            await tr.update(targetUserRef, { Money: state.targetUserMoney })
```

また、オブジェクトのプロパティ名が先頭大文字のものがありますが、先頭小文字で統一するのが一般的です。